File: CodingInterviewsPackSolutionsCPP/Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int sz = nums.size();
        for(int start = 0, end = 0; end < sz; end++){
            if(nums[end]!=0)
                swap(nums[start++],nums[end]);
        }
    }
};

File: CodingInterviewsPythonSolutions/Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        start = 0
        end = 0
        sz = len(nums)
        
        while end < sz:
            if nums[end] != 0:
                nums[start], nums[end] = nums[end], nums[start]
                start += 1
            end += 1
